### PR TITLE
Add option to request spot instances on AWS

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

```release-note
Add an option to create spot instances instead of normal instances on AWS
```

/assign @mrIncompetent

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -459,8 +462,14 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloud.MachineCreateDe
 		})
 	}
 
+	var instanceMarketOptions *ec2.InstanceMarketOptionsRequest
+	if config.IsSpotInstance != nil && *config.IsSpotInstance {
+		instanceMarketOptions = &ec2.InstanceMarketOptionsRequest{MarketType: aws.String("spot")}
```
Comment (mrIncompetent) [inline pkg/cloudprovider/provider/aws/provider.go]: nit: Theres a const for "spot" in the ec2 package: 
```go 
const (
	// MarketTypeSpot is a MarketType enum value
	MarketTypeSpot = "spot"
)
```

Do we also want to allow for configuration of `BlockDurationMinutes` ?

Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: SGTM and I was grepping for `Spot` not for `spot` as that is what they used in their example: https://aws.amazon.com/blogs/compute/new-amazon-ec2-spot-pricing/

> Do we also want to allow for configuration of BlockDurationMinutes ?

I guess thats reasonable. If and where we actually usage that option later on is not something we have to settle on right now

**Comment by mrIncompetent** [comment]
I've introduced the const for "spot" and rebased.

/lgtm
/approve

/hold
@alvaroaleman any objections?

**Comment by alvaroaleman** [comment]
No

/hold cancel

**Comment by mrIncompetent** [comment]
Still the hetzner apt mirror issues

**Pull request merged**